skill-tree /src/test/java/com/RDS/skilltree/SkillsIntegrationTests.java

```diff
@@ -1,137 +1,250 @@
-                .body("data[0].id", equalTo("s1"))
-                .body("data[0].name",equalTo("Java"))
-                .body("data[0].type", anyOf(equalTo("ATOMIC"), equalTo("DERIVED")))
-                .body("data[0].description",isA(String.class));
+                .body("id", equalTo(String.valueOf(skillId)))
+                .body("name", equalTo("Java"))
+                .body("type", equalTo("ATOMIC"));
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return 400 on incorrect userId")
-    public void testAPIReturns400_OnIncorrectUserId() {
+    @DisplayName("Return 404, on skill not found given SkillId")
+    public void testAPIReturns404_OnSkillNotFound() {
+        UUID skillId = UUID.randomUUID();
         Response response = given()
-                .get("/skills/{userId}");
+                .pathParam("id", skillId)
+                .get("/v1/skills/{id}");
 
         response.then()
-                .statusCode(400)
-                .body("code",equalTo(400))
-                .body("message",equalTo("Invalid userId passed"));
+                .statusCode(404)
+                .body("message", equalTo("Skill not found with given Id"));
 
     }
 
     @Test
-    @Disabled
-    @DisplayName("Return 404 on, no skills found on given userId")
-    public void testAPIReturns404_OnNoSkillsFound() {
+    @DisplayName("Return 200, on skill with given name")
+    public void testAPIReturns200_OnSkillFoundGivenName() {
         Response response = given()
-                .get("/skills/{userId}");
+                .pathParam("name", "Java")
+                .get("/v1/skills/name/{name}");
+
+        response.then()
+                .statusCode(200)
+                .contentType("application/json")
+                .body("name", equalTo("Java"))
+                .body("type", equalTo("ATOMIC"));
+    }
+
+    @Test
+    @DisplayName("Return 404, if skill given skill name is not found")
+    public void testAPIReturns404_OnSkillGivenSkillNameNotFound() {
+        Response response = given()
+                .pathParam("name", "Go")
+                .get("/v1/skills/name/{name}");
+
 
         response.then()
                 .statusCode(404)
-                .body("code", equalTo(404))
-                .body("message",equalTo("No skills associated with the user"));
+                .body("message", equalTo("Skill not found with the given name"));
+    }
 
+    @Test
+    @DisplayName("Return 400, if createdBy is not passed for Skill creation")
+    public void testAPIReturns400_OnCreatedByNotPassedForSKillCreation() {
+        SkillDRO skillDRO = SkillDRO.builder()
+                .name("Go")
+                .type(SkillType.ATOMIC)
+                .build();
+
+        Response response = given()
+                .contentType("application/json")
+                .body(skillDRO)
+                .post("/v1/skills/");
+
+        response.then()
+                .statusCode(400)
+                .body("error", equalTo("Bad Request"));
```

Comment:
can we also provide a message so that the client has a clue what went wrong

Reply:
This will be handled in exception handling changes which I will do in coming time.

Reply:
Will check this in this ticket - https://github.com/Real-Dev-Squad/skill-tree-backend/issues/31